File: src/Entity/Produit.php
<?php

namespace App\Entity;
use Symfony\Component\Form\Extension\Core\Type\FileType;
use App\Repository\ProduitRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
/**
 * @ORM\Entity(repositoryClass=ProduitRepository::class)
 */
class Produit
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $idproduit;

    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     *@Assert\NotBlank(message="libelle obligatoire")
     */
    private $Libelle;
    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     *@Assert\NotBlank(message="description obligatoire")
     */
    private $Description;


    /**
     * @ORM\Column(type="float", nullable=true)
     * @Assert\NotBlank(message="prix obligatoire")
     */
    private $prix;

    /**
     * @ORM\Column(type="string", length=255, nullable=false)
     */
    private $image;

    /**
     * @ORM\ManyToOne(targetEntity=Categorie::class, inversedBy="Produit")
     * * @Assert\NotBlank(message="categorie du produit obligatoire")
     */
    private $categorie;


    public function getIdproduit(): ?int
    {
        return $this->idproduit;
    }

    public function getLibelle(): ?string
    {
        return $this->Libelle;
    }

    public function setLibelle(?string $Libelle): self
    {
        $this->Libelle = $Libelle;

        return $this;
    }
    public function getDescription(): ?string
    {
        return $this->Description;
    }

    public function setDescription(?string $Description): self
    {
        $this->Description= $Description;

        return $this;
    }
    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(?float $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getCategorie(): ?Categorie
    {
        return $this->categorie;
    }

    public function setCategorie(?Categorie $categorie): self
    {
        $this->categorie = $categorie;

        return $this;
    }
}

File: codename-project/src/com/mycomany/entities/Produit.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycomany.entities;

/**
 *
 * @author lenovo
 */
public class Produit {
     public int idproduit ; 
    public String libelle; 
    public String description; 
    public String image;   
    public Categorie categorie;

    public Produit(String libelle, String description, Categorie categorie, int prix) {
        this.libelle = libelle;
        this.description = description;
        this.categorie = categorie;
        this.prix = prix;
    }
    public int prix;    
 
    public Produit() {
        
    }
      public Produit( String libelle, String description, String image, Categorie categorie, int prix) {
        
        this.libelle = libelle;
        this.description = description;
        this.image = image;
        this.categorie = categorie;
        this.prix = prix;
    }

  

  

    @Override
    public String toString() {
        return "Produit{" + "idproduit=" + idproduit + ", libelle=" + libelle + ", description=" + description + ", image=" + image + ", categorie=" + categorie + ", prix=" + prix + '}';
    }

    public Produit(String libelle, String description, int prix) {
        this.libelle = libelle;
        this.description = description;
        this.prix = prix;
    }

 

    

    public int getIdproduit() {
        return idproduit;
    }

    public Categorie getCategorie() {
        return categorie;
    }

    public void setCategorie(Categorie categorie) {
        this.categorie = categorie;
    }

    public void setIdproduit(int idproduit) {
        this.idproduit = idproduit;
    }

    public String getLibelle() {
        return libelle;
    }

    public void setLibelle(String libelle) {
        this.libelle = libelle;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

  

    public int getPrix() {
        return prix;
    }

  

    public void setPrix(int prix) {
        this.prix = prix;
    }

    public Produit(int idproduit, String libelle, String description, String image, int prix) {
        this.idproduit = idproduit;
        this.libelle = libelle;
        this.description = description;
        this.image = image;
       
        this.prix = prix;
    }




    
}
